### ISSUE TITLE
Provide a way to remove 'message filters'

### ISSUE THREAD

**Issue by museun** [museun/twitchchat#8]
The `on` method should return a unique `Id` that can be stored and used later to remove that filter.
While this is being done, determine whether the name should be a combination of `on/off` or `filter/remove_filter`

**Comment by museun** [comment]
I think there should be something as follows:
```rust
#[derive(Clone, Copy, PartialEq)]
struct Id(u16);

#[derive(Default)]
struct IdGen(u16);

impl IdGen {
    fn new() -> Self {
        Self::default()
    }

    fn next(&mut self) -> Id {
        let next = self.0;
        self.0 += 1;
        Id(next)
    }
}


// [ pseudo ]
impl Client {
    fn filter(&mut self, ..) -> Id {
        self.idgen.next()
    }

    fn remove_filter(&mut self, id: Id) -> bool {
        self.filters.remove(id)
    }
}
```

a `u16` should be enough, a u8 could work if a reclamation scheme was implemented
`remove_filter` should return whether this id was still valid

this id then could be used in logging

**Comment by museun** [comment]
fixed in f6e11d6
